### DISCUSSION TITLE
Move/Swap selection and move next/previous word or symbol

### DISCUSSION THREAD

**Discussion by pathwave** [helix-editor/helix#3981] [Ideas] [answered]
Hi,

I have two ideas/features that atleast I personally would like to have.

The first one is a command to move/swap selection followed by a movement:
For example if we have the text "foo bar", and currently the word "foo" is selected, one could do the "swap_selection" command followed by "move_next_word_*". The result is then that the current selection "foo" gets swapped with what the next movement selects, in this case "bar". so the resulting text ends up being "bar foo" (with foo still selected).

The other idea is a simple movement command that moves to the next word or symbol, whatever comes first.

With this example text: Foo(Bar(""))
Using "move_next_word_*" we get the following stops: Foo, (, Bar, (""))
With the new command the stops would be: Foo, (, Bar, (, ", " , ), )

I opened a discussion here because I was curious if this is something worth putting into the editor or if I should wait for a plugin system, or maybe somebody knows of a better way

**Comment by AlexanderBrevig** [comment, marked as answer]
Hi 👋🏻 
For your first idea, check https://docs.helix-editor.com/keymap.html#selection-manipulation and `A-(` `A-)`. What would it mean to `swap k` or `swap p`? I think `(` and `)` represents the relevant motions for this rotation.

The second idea I'm curious as to why you would want it. Have you tried using tree sitter to navigate? `A-n` `A-p` `A-i` `A-o`.

**Reply by pathwave** [reply]
Hi,
A-( and A-) seems to be what I am after! I just need to find a way to do it the way I want :)
I will look into the tree sitter navigation aswell :+1: 

Thank you
